src/prehandlers/integrationConfigPrehandler.ts

```diff
@@ -0,0 +1,75 @@
+import type {
+  CONNECTOR_REQUEST_HEADERS_SCHEMA,
+  CONNECTOR_REQUEST_HEADERS_SCHEMA_PRE_SETUP,
+} from '@lokalise/connector-api-contracts'
+import type { ApiContractMetadataToRouteMapper } from '@lokalise/fastify-api-contracts'
+import { PublicNonRecoverableError } from '@lokalise/node-core'
+import type { FastifyReply, FastifyRequest, HookHandlerDoneFunction } from 'fastify'
+import type { z } from 'zod/v4'
+import { decodeBase64 } from '../utils/base64Utils.ts'
+
+export const PROTECTED_ROUTE_METADATA_MAPPER = (() => {
```

Comment:
I'm not sure if this is the best name, alternate suggestions would be most welcome!